src/simulation/LoadGenerator.cpp

```diff
@@ -1804,6 +1958,43 @@ LoadGenerator::checkAccountSynced(Application& app, bool isCreate)
     return result;
 }
 
+bool
+LoadGenerator::checkMinimumSorobanInvokeSuccess(GeneratedLoadConfig const& cfg)
+{
+    if (!cfg.modeInvokes())
+    {
+        // Only applies to soroban invoke transactions
+        return true;
+    }
+
+    auto const& invokeCfg = cfg.getSorobanInvokeConfig();
+    if (invokeCfg.minPercentSuccess == 0)
+    {
+        // Vacuously true
+        return true;
+    }
+
+    // Count the total number of invoke transactions included in blocks, as well
+    // as the number of those transactions that were successful.
+    size_t nInvokes = 0;
+    size_t nSuccessful = 0;
+    for (auto const& hash : mInvokeTxHashes)
+    {
+        std::optional<TransactionResult> txr =
+            loadTransactionResult(mApp.getDatabase(), hash);
```

Comment:
An idea for simplification here: we can use metrics to decide loadgen exit status (for example, we have `ledger.apply.success` and `ledger.apply.failure` that we can compare before and after the run). This way, we don't need to do any expensive database queries, plus loadgen wouldn't need to know about any of the tx result internals

Reply:
Good idea. The tradeoffs as I see them are:

* If there are other non-loadgen transactions during the loadgen run, those will interfere with these metrics. That being said, I don't think we ever run loadgen simultaneously with non-loadgen traffic.
* This requires changing the `minpercentsuccess` option to cover *all* transactions rather than just `SOROBAN_INVOKE` transactions. I think that's fine, it's easy enough to translate from one to the other when taking the blend of transaction types into account.

All in all I think it's worth the tradeoffs to avoid the database calls and dependence on the `txhistory` table structure. I'll make this change.

Reply:
> If there are other non-loadgen transactions during the loadgen run, those will interfere with these metrics. That being said, I don't think we ever run loadgen simultaneously with non-loadgen traffic.

Agreed, in practice having other transactions during loadgen run invalidates the results of the run anyway, so I think it's safe to assume we won't have any other transactions.

> This requires changing the minpercentsuccess option to cover all transactions rather than just SOROBAN_INVOKE transactions. I think that's fine, it's easy enough to translate from one to the other when taking the blend of transaction types into account.

I might be missing something, but I wonder why this part needs to change?

Reply:
> I might be missing something, but I wonder why this part needs to change?

It's fine when all transactions are `SOROBAN_INVOKE`, but in blended mode these counts include all transaction types (not just `SOROBAN_INVOKE`) so it's not possible to figure out exactly which ones failed without checking the database.

Reply:
I see. We could also split metrics for classic vs soroban (actually, it'd be really useful even outside of this change as right now we don't have enough visibility). Then the `minpercentsuccess` option would only be applicable to soroban transactions.

Reply:
Good idea. I split the metrics and used the soroban-specific metrics to check `minpercentsuccess` (see fcb48f779fd897681a5c84096a8fd1f558627773).

Does anything else outside this repo need updating with the new metrics names (Grafana, for example)?

Reply:
Thanks! Yeah, ideally, we can also update our grafana boards. Since you've only added new metrics and didn't rename any old ones, this change is non-breaking, so we can merge it first, then update Grafana.